tests/infra/clients.py

```diff
@@ -423,6 +468,8 @@ def __init__(self, *args, **kwargs):
         if os.getenv("HTTP"):
             if os.getenv("CURL_CLIENT"):
                 self.client_impl = CurlClient(*args, **kwargs)
+            elif os.getenv("WEBSOCKETS_CLIENT"):
+                self.client_impl = WSClient(*args, **kwargs)
```

Comment:
I don't think this should be a separate client? Presumably we just have .upgrade() on our existing requests-based client?

Reply:
Hmmm, not sure about this one. As I see it, clients would _not_ use a http client for some requests, then upgrade to ws. Instead, I think they would connect to `wss://node0.ccf.io:55555` from the start and the endpoint would be upgraded to ws straight away. 

Also, AFAICS Python's `requests` does not support upgrading to ws so we would need to create a new endpoint in the enclave on upgrade (i.e. when `create_connection()` is called).

Reply:
Ah fair enough then.